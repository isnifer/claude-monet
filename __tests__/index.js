import 'react-native'
import React from 'react'
import Monet, { withTheme } from '../'
import renderer from 'react-test-renderer'
import Head from '../examples/Head'
import '../examples/themes/C3PO'

test('Renders <Head /> w/out withTheme', () => {
  const tree = renderer.create(
    <Head />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Renders <Head /> with Head theme', () => {
  const ThemedComponent = withTheme('Head')(Head)
  const tree = renderer.create(
    <ThemedComponent />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Renders <Head /> with Head:Green modifier', () => {
  const ThemedComponent = withTheme('Head', ['Green'])(Head)
  const tree = renderer.create(
    <ThemedComponent />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
