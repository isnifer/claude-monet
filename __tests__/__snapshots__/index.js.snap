// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Head /> w/out withTheme 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "purple",
      }
    }
  >
    Head Component
  </Text>
</View>
`;

exports[`Renders <Head /> with Head theme 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "gold",
      }
    }
  >
    Head Component
  </Text>
</View>
`;

exports[`Renders <Head /> with Head:Green modifier 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "green",
      }
    }
  >
    Head Component
  </Text>
</View>
`;
